tests/test_fsclient.py

```diff
@@ -134,6 +133,22 @@ def test_insert_one(make_db, rc, input, result):
     assert list(client.dbs[dbname][collname].values())[-1] == result
 
 
+test_insert_cif_bad = [{'bad_case_test_dict': 'bad'}, 'bad_case_test_str']
+def test_insert_one_bad(rc):
+    client = FileSystemClient(rc)
+    client.open()
+
+    dbname = 'local'
+    collname = 'calculated'
+
+    client.load_database(pydr_rc['databases'][0])
+
+    with pytest.raises(KeyError):
```

Comment:
do we want to return more info back to the user, for example, printing what the bad entry was?  To help with debugging?  I don't remember the syntax but there is a way of capturing the error message and doing an assert on it.